### PR TITLE
Demo rfc80 density plot

### PR DESCRIPTION
**Pull request by uklineale**
Fix # (see https://help.github.com/en/articles/closing-issues-using-keywords)

Describe changes proposed in this pull request:
- Use clickhouse for density plot calls
- Refactor logic into service to clean up controller
- Simplify logic into Java streams

# Checks
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). We can fix this during merge by using a squash+merge if necessary
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Make sure your PR has one of the labels defined in https://github.com/cBioPortal/cbioportal/blob/master/.github/release-drafter.yml

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [Giphy CAPTURE](https://giphy.com/apps/giphycapture) or [Peek](https://github.com/phw/peek)

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

File: src/main/java/org/cbioportal/service/impl/ClinicalDataDensityPlotServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            result.setPearsonCorr(0.0);
+        }
+
+        // filter out empty bins
+        result.setBins(result.getBins().stream().filter((bin)->(bin.getCount() > 0)).collect(Collectors.toList()));
+        return result;
+    }
+
+    @Override
+    public List<ClinicalData> filterClinicalData(List<ClinicalData> clinicalDataList) {
```
Comment (alisman) [inline src/main/java/org/cbioportal/service/impl/ClinicalDataDensityPlotServiceImpl.java:136]: not sure if i understand, this is not yet wired up with clinical filters, right?

File: src/main/java/org/cbioportal/service/ClinicalDataDensityPlotService.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package org.cbioportal.service;
+
+import org.cbioportal.model.ClinicalData;
+import org.cbioportal.model.DensityPlotData;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+public interface ClinicalDataDensityPlotService {
+    DensityPlotData getDensityPlotData(List<ClinicalData> filteredClinicalData,
```
Comment (haynescd) [inline src/main/java/org/cbioportal/service/ClinicalDataDensityPlotService.java:10]: Can we Take these arguments and throw them into a POJO (maybe using a builder pattern) so we aren't passing so many  args to the fn

Comment (uklineale) [reply src/main/java/org/cbioportal/service/ClinicalDataDensityPlotService.java:10]: good idea

File: src/main/java/org/cbioportal/service/ClinicalDataDensityPlotService.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                                       Boolean xAxisLogScale,
+                                       Boolean yAxisLogScale,
+                                       Integer xAxisBinCount,
+                                       Integer yAxisBinCount,
+                                       BigDecimal xAxisStart,
+                                       BigDecimal yAxisStart,
+                                       BigDecimal xAxisEnd,
+                                       BigDecimal yAxisEnd);
+    
+    List<ClinicalData> filterClinicalData(List<ClinicalData> clinicalDataList);
```
Comment (haynescd) [inline src/main/java/org/cbioportal/service/ClinicalDataDensityPlotService.java:22]: You don't need this fn if you just do everything in the getDensityPlotData Fn

File: src/main/java/org/cbioportal/service/impl/ClinicalDataDensityPlotServiceImpl.java (hunk 1 above)
Comment (uklineale) [reply src/main/java/org/cbioportal/service/impl/ClinicalDataDensityPlotServiceImpl.java:136]: I used the wrong name, hence your confusion. This basically extracts only values that can be used in the density plot. It just removes samples that don't have data for botht he X and Y axis, as well as removes non-numeric ClinicalData.

File: src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java [hunk 4]
```diff
@@ -41,16 +73,39 @@
...
+    @Autowired
+    private PatientServiceImpl patientService;
+    @Autowired
+    private StudyViewFilterApplier studyViewFilterApplier;
+    @Autowired
+    private StudyViewFilterUtil studyViewFilterUtil;
+    @Autowired
+    private SampleServiceImpl sampleService;
+    @Autowired
+    private ClinicalDataFetcher clinicalDataFetcher;
```
Comment (haynescd) [inline src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java:91]: Are these needed?

File: src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java [hunk 5]
```diff
@@ -41,16 +73,39 @@
...
+    @Autowired
+    private ClinicalDataDensityPlotService clinicalDataDensityPlotService;
```
Comment (haynescd) [inline src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java:93]: You don't need @Autowired annotation. Just set it to final and set it via constructor dep injection

**Review by haynescd — APPROVED** [review]
LGTM

**Comment by haynescd** [comment]
Great Job @uklineale

**Pull request merged**